divert/core.py

```diff
@@ -16,6 +18,19 @@ def raise_again(self, *_: Any) -> bool:
         self._jumps_remaining -= 1
         return self._jumps_remaining > 0
 
+    @classmethod
+    def make_weak(cls, *args: Any, **kwargs: Any) -> Diversion:
+        """Make weak variant of the diversion class.
+
+        By dynamically forming it provides support for generic extensions of the base diversions.
+        The minor performance hit is acceptable vs. having to maintain `WeakDiversion` instances for each extension.
+        """
+
+        class WeakDiversion(cls, Exception):
```

Comment:
one weakness with dynamically creating this class is that it is unique every time `make_weak` is called. That could happen if it's invoked in multiple modules, which might happen if someone is hoping to manually catch a diverted payload across modules.
given:
```
>>> def f():
...     class C(Exception): ...
...     return C
>>> A = f()
>>> B = f()
# If the same instance is used:
>>> try:
...     raise A()
... except A:
...     pass
>>>
# if a different instance is used:
>>> try:
...     raise A()
... except B:
...     pass
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
__main__.f.<locals>.C
```
You can get around this by caching the result of this function (though be careful to access the value on the _class itself_)
```
>>> class E(Exception):
...     weakened = None
...     @classmethod
...     def make_weak(cls):
...             if cls.weakened is None:
...                     class Weakened(cls, Exception): ...
...                     cls.weakened = Weakened
...             return cls.weakened
>>> A = E.make_weak()
>>> B = E.make_weak()
>>> try:
...     raise A()
... except B:
...     pass
>>>
```

Reply:
True, ya, good call